File: DoctorsListAPI/Models/Doctor.cs
using System.Collections.Generic;

namespace DoctorsListAPI.Models
{
    public class Doctor
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Gender { get; set; }
        public MedicalSchool MedicalSchool { get; set; }
        public ICollection<DoctorSpecialty> DoctorSpecialties { get; set; }
        public ICollection<DoctorLanguage> DoctorLanguages { get; set; }
        public ICollection<PatientRating> PatientRatings { get; set; }
    }
}

File: DoctorListWeb/src/app/models/doctor.ts
import { DoctorSpecialty } from './doctorSpecialty';
import { DoctorLanguage } from './doctorLanguage';
import { PatientRating } from './patientRating';

export interface Doctor {
  id: number;
  name: string;
  gender: string;
  specialties?: string;
  averageRating?: number;
  medicalSchool?: string;
  doctorSpecialties?: DoctorSpecialty[];
  doctorLanguages?: DoctorLanguage[];
  patientRatings?: PatientRating[];
}
